mkosi/distribution/opensuse.py

```diff
@@ -261,6 +256,24 @@ def is_kernel_package(cls, package: str) -> bool:
         return package in ("kernel-default", "kernel-kvmsmall")
 
 
+def fetch_gpgkeys(context: Context) -> list[Path]:
+    files = set()
+
+    for p in (context.sandbox_tree / "etc/zypp/repos.d").iterdir():
+        for _, name, value in parse_ini(p):
+            if name != "gpgkey":
+                continue
+
+            keys = value.splitlines()
+            for key in keys:
+                if not key.startswith("file://"):
+                    continue
+
+                files.add(Path(key.removeprefix("file://")))
+
+    return list(files)
```

Comment:
sets still don't guarantee order
```suggestion
    return sorted(files)
```
or make `files` a dict and do
```python
files[Path(key.removeprefix("file://"))] = None
```
then `list(files.keys())` will be in insertion order.